cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -419,19 +425,33 @@ def prepare_download(self, file_list, file_name, scratch_dir):
         return tmp_dir, file_name
 
     def download_products(self,):
-        #print('in url file_list', self.args.get('file_list'))
-        #scratch_dir, logger = set_session(self.args.get('session_id'))
-
-        file_list = self.args.get('file_list').split(',')
-        #print('used file_list', file_list)
-        file_name = self.args.get('download_file_name')
-
-        tmp_dir, target_file = self.prepare_download(
-            file_list, file_name, self.scratch_dir)
-        #print('downlaoding scratch dir', self.scratch_dir)
         try:
+            # TODO not entirely sure about these
+            self.off_line = False
+            self.api = False
+
+            if not self.public:
```

Comment:
and if not public, data from any job_id can be downloaded?

Reply:
Adding also this check

Reply:
So, in case the request is public, a check of the provided `job_id`, against the one calculated with the analysis_parameter from the scratch directory (and without any token since this is not sent) is performed.

Reply:
Does it make sense?